src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,57 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int LENGTH = 10;
+    private V[] value;
+    private K[] key;
+    private int index;
+
+    public StorageImpl() {
+        value = (V[]) new Object[LENGTH];
+        key = (K[])new Object[LENGTH];
+    }
+
     @Override
     public void put(K key, V value) {
+        boolean keyWas = false;
+        for (int i = 0; i < index; i++) {
+            if (compare(this.key[i], key)) {
+                this.value[i] = value;
+                keyWas = true;
+            }
+        }
```

Comment:
I'd recommend something like this
```suggestion
        for (int i = 0; i < index; i++) {
            if (compare(this.key[i], key)) {
                this.value[i] = value;
                return
            }
        }
```